cdap-security/src/main/java/co/cask/cdap/security/server/ExternalAuthenticationServer.java

```diff
@@ -217,8 +218,8 @@ public String getName() {
 
       @Override
       public InetSocketAddress getSocketAddress() throws RuntimeException {
-        if (announceAddress != null) {
-          return new InetSocketAddress(announceAddress, connector.getLocalPort());
+        if (announceAddressDeprecated != null) {
+          return new InetSocketAddress(announceAddressDeprecated, connector.getLocalPort());
```

Comment:
Don't do this. This is exactly what the `CDAP-4535` is about. If you register with this discoverable, when multiple auth server are started, one of it will hang.